src/testing/anndata/_pytest.py

```diff
@@ -32,10 +32,14 @@ def pytest_configure(config: pytest.Config) -> None:
 
 
 @pytest.fixture(autouse=True)
-def _suppress_env_for_doctests(request: pytest.FixtureRequest) -> None:
+def _anndata_test_env(request: pytest.FixtureRequest) -> None:
+    import anndata
+
     if isinstance(request.node, pytest.DoctestItem):
         request.getfixturevalue("_doctest_env")
 
+    anndata.settings.reset(anndata.settings._registered_options.keys())
+
```

Comment:
Not sure how I feel about a behavioral change like this being used on every test....took me a second to find this after seeing the lack of context manager.  So I'm not sure how easy this will be for the next person.   I won't block on it though.  I see the argument to be made that `settings` has a sort of implicit `function` scope i.e., changing in one test shouldn't change in another.  Could be nice to start a document listing all the custom things we do since they can be hard to find for people new to the repo

Reply:
That would indeed be nice to have.

I think it’s very important to do all these resets in the tests (scanpy has many more) since they
1. document what global state your package has (and usually you don’t want it to have much of that)
2. make sure the tests are actually correct and no leakage from other tests causes them to be order-dependent. Another way to help with that would be to use something like https://pypi.org/project/pytest-randomly/ to randomize test order

Reply:
Sounds very reasonable to me.  Good call then.